Add database

goloviu opened the pull request.

goloviu committed:
feat: реализовать базу данных in-memory H2
feat: добавить новые контроллеры
GenreController, RatingController
fix: изменить логику по добавлению в друзья
feat: добавить новые сущности FilmGenre, FilmRating
delete: удалить старую реализацию хранилищ

goloviu committed:
style: переместить запятую

goloviu committed:
style: переместить запятые

goloviu committed:
style: переместить запятые

goloviu committed:
style: переместить запятую

goloviu committed:
style: переместить запятую

goloviu committed:
fix: убрать IGNORE в data.sql

goloviu committed:
Update README.md

goloviu committed:
Merge branch 'main' into add-database

dgumich commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE IF NOT EXISTS users (id int4 GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,name varchar,email varchar NOT NULL UNIQUE,login varchar NOT NULL UNIQUE,birthday date NOT NULL)
```
Да, компактно, но не читаемо=)
Не поленись и напиши с новой строчки каждый столбец.
Кроме того для varchar укажи количество символов, сколько должно быть (varchar(10) или varchar(129))

goloviu replied:
Исправлено

goloviu committed:
Дополнительные изменения

fix: ограничить количество символов в sql скрипте по varchar
fix: изменить вызов sql запросов в цикле на более быстрый метод barchUpdate
fix: убрать аннотации валидации FilmGenre FilmRating
fix: дополнить валидацию isValidFilm проверкой на жанры и рейтинг
style: привести скрипт sql в читаемый вид

goloviu committed:
Merge branch 'add-database' of https://github.com/goloviu/java-filmorate into add-database

goloviu committed:
style: удалить неиспользуемые импорты

goloviu committed:
style: удалить неиспользуемые импорты в тестах

goloviu committed:
fix: добавить INSERT IGNORE в sql скрипт

goloviu committed:
fix: удалить несовместимый INSERT IGNORE с тестами проекта

goloviu committed:
fix: увеличить количество допустимых символов жанра в sql скрипте

goloviu committed:
fix: исправление тестов

dgumich commented on src/main/java/ru/yandex/practicum/filmorate/exceptionhandler/ErrorHandler.java:
```diff
@@ -40,17 +42,17 @@ public ErrorResponse userExistHandle(final UserAlreadyExistException exception)
+    @ExceptionHandler({EmptyResultDataAccessException.class, UserNotFoundException.class})
     @ResponseStatus(HttpStatus.NOT_FOUND)
-    public ErrorResponse userNotFoundHandle(final UserNotFoundException exception) {
+    public ErrorResponse userNotFoundHandle(final Throwable exception) {
```
Обычно пытаются сузить тип параметра как могут. Throwable слишком общий. Хотя бы до RunTimeException

dgumich commented on src/main/java/ru/yandex/practicum/filmorate/storage/user/UserDbStorage.java:
```diff
@@ -0,0 +1,167 @@
+import java.util.List;
+import java.util.Set;
+
+@Component("userDbStorage")
```
Молодец, что даешь название бину. Хотя, спринг по умолчанию даст точно такое же, на некоторых проектах это делают в любом случае.

goloviu committed:
fix: сделать валидацию FilmGenre FilmRating через @Max

goloviu committed:
fix: удалить неиспользуемые импорты
